pandas/tests/indexes/multi/test_setops.py

```diff
@@ -760,4 +764,4 @@ def test_union_with_na_when_constructing_dataframe():
     series2 = Series((10, 20), index=MultiIndex.from_tuples(((None, None), ("a", "b"))))
     result = DataFrame([series1, series2])
     expected = DataFrame({(np.nan, np.nan): [1.0, 10.0], ("a", "b"): [np.nan, 20.0]})
-    tm.assert_frame_equal(result, expected)
+    tm.assert_frame_equal(result, expected, check_column_type=False)
```

Comment:
Is this because of the string inference in `MultiIndex.from_tuples`?

Reply:
Yep, the dtypes of result are correct here

Reply:
Gotcha. Could we just make the MultiIndex from arrays so we can avoid this? (the MI creation doesn't look critical for the tested behavior)